torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3249,6 +3249,25 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::endCoalescing() {
   return endCoalescing(OpType::COALESCED);
 }
 
+void ProcessGroupNCCL::startCommTimeEstimate() {
+  groupStart();
+}
+
+float ProcessGroupNCCL::endCommTimeEstimate() {
+#ifdef NCCL_SIM_INFO_INITIALIZER
+  auto simInfo = std::make_shared<ncclSimInfo_t>();
```

Comment:
why using a shared_ptr when we have no intention to share? (`simInfo` dies within this function's scope).
At `ncclCollectiveEstimateEnd`, I wonder if we can directly return the time.